std/algorithm/comparison.d

```diff
@@ -1911,3 +1915,97 @@ bool isPermutation(alias pred = "a == b", Range1, Range2)
         [mytuple(2, 3), mytuple(1, 2)]
     ));
 }
+
+enum implicitlyConvertsToBool(T) = isImplicitlyConvertible!(bool, T) || isArray!T;
```

Comment:
Ok. Done. 

I still need to define it as

enum ifTestable(T) = __traits(compiles, { if ((T())) {} }) || isArray!T;

otherwise compilation fails as

```D
comparison.d(1993,18): Error: template std.algorithm.comparison.either cannot deduce function from argument types !()(string, string), candidates are:
comparison.d(1938,15):        std.algorithm.comparison.either(alias pred = (a) => a, Ts...)(lazy Ts a) if (a.length > 1 && !is(CommonType!Ts == void) && allSatisfy!(ifTestable, Ts))
comparison.d(1951,10):        std.algorithm.comparison.either(alias pred = (a) => a, Ts...)(ref Ts a) if (a.length > 1 && allSameType!Ts && allSatisfy!(ifTestable, Ts))
```

Any idea why?